### PR TITLE
docker: fix docker image to suit storage in memory

### LINKED ISSUE #80005: Scripts from `docker-entrypoint-initdb.d` are not executed when using `--store=type=mem,size=0.25` for `start-single-node`
We have successfully been using an adapted Docker image with a file in `/docker-entrypoint-initdb.d` to create a user and apply the suggested testing configuration: https://github.com/prisma/engine-images/blob/7695bfcc990b5f91cf5d86b15ec95fc3b129f709/cockroach/cockroach.Dockerfile

Now we were also recommended to use `--store=type=mem,size=0.25` as an additional argument to `start-single-node --insecure` and noticed that a) the user is not present any more and b) all the configuration is default - so I think our `.sql` file from our image is not run correctly.

Reproduction is possible via this docker-compose:
```
version: "3"
services:
  test-db-cockroachdb:
    image: prismagraphql/build:cockroach-custom
    restart: always
    command: start-single-node --insecure
    ports:
      - "26257:26257"
    networks:
      - engine-tests

  test-db-cockroachdb-mem:
    image: prismagraphql/build:cockroach-custom
    restart: always
    command: start-single-node --insecure --store=type=mem,size=0.25
    ports:
      - "26258:26257"
    networks:
      - engine-tests

networks:
  engine-tests: null
```

The `prismagraphql/build:cockroach-custom` is our customized image that adds the `.sql` file to `cockroachdb/cockroach-unstable:v22.1.0-alpha.1`, it is built via https://github.com/prisma/engine-images/blob/7695bfcc990b5f91cf5d86b15ec95fc3b129f709/cockroach/cockroach.Dockerfile linked above.

### PR DESCRIPTION
**Pull request by ZhouXing19**
Currently the datasets created in the init process are stored on disk,
and they are inaccessible if the “real” server is started with storage in memory.

This is because, for the container initialization, the current logic is:
1. start_init_server: start the single-node server in the background, with
default on-disk storage setting;
2. run_init_queries: create default db and user, and run the init scripts
saved in
docker-entrypoint-initdb.d;
3. *stop the init server and restart the server* with given parameters in foreground.

In this commit we start and keep running the init server with the given
parameters, and eliminate the “stop and restart” step. Also, bring the server
process from background to foreground with `fg` once all init queries are finished.

Fixes #80005.

Release note (docker): refactor the initialization process of the docker image
to accomodate initialization scripts with memory storage.

### CODE REVIEW

File: pkg/testutils/docker/single_node_docker_test.go [hunk 1]
```diff
@@ -499,17 +523,17 @@ func (dn *dockerNode) waitServerStarts(ctx context.Context) error {
...
-		return errors.Wrapf(err, "cannot run fsnotify to listen to %s:\nres:%#v", listenURLFile, res)
+		return errors.Wrapf(err, "cannot run fsnotify to listen to %s:\nres:%#v", initSuccessFile, res)
 	}
 
-	if strings.Contains(res.stdOut, "finished\r\n") {
+	if strings.Contains(res.stdOut, "finished") {
 		return nil
 	}
-	return errors.Wrap(errors.Newf("%s", res), "error in waiting the server to start")
+	return errors.Wrap(errors.Newf("%#v", res), "error in waiting the initialization to finish")
```
Comment (otan) [inline pkg/testutils/docker/single_node_docker_test.go:536]: nit: `errors.New("error waiting for initialization to finish: %v", err)`

**Review by otan — APPROVED** [review]
Lookgs good. You'll need to fix up the release note + remove [WIP] from the commit message.

We'll also want to backport this.

File: build/deploy/cockroach.sh [hunk 2]
```diff
@@ -257,64 +251,28 @@ create_default_user() {
...
     process_init_files /docker-entrypoint-initdb.d/*
-    # Stop the init server.
-    stop_init_node
-    touch init_success
+    # Bring the background server process to the foreground, otherwise the
+    # docker container will automatically exit here.
+    echo "init_finished" > ./cockroach-data/init_success
+    fg %1
+  else
+      # Start the real server.
```
Comment (rafiss) [inline build/deploy/cockroach.sh:269]: "real server" is probably not a necessary comment -- since the `if` branch above is also a "real" server now

Comment (ZhouXing19) [reply build/deploy/cockroach.sh]: Done.

**Review by rafiss — APPROVED** [review]
lgtm!

**Comment by ZhouXing19** [comment]
Thanks!

bors r+

**Pull request merged**